balancer-labs/frontend-v2
# Refactor fNum to allow for Internationalisation / Multiple Currencies

# Description

I've created a new function [fNum2](https://github.com/balancer-labs/frontend-v2/pull/1321/files#diff-8b279947dd8a9ce4af0f314eab942df05750b46dbc2aa4fe71b65d00d33c46a6R90) that uses Javascripts native [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) to format numbers instead of the old numeral package. 

Nothing should be changed UX wise in this PR. It's purely a code refactor. To ensure nothing has changed I created a series of tests that have an old fNum call and an fNum2 equivalent that runs through many test cases to ensure the results are always the same. 

fNum2 has several advantages over the old fNum:

- Users can change their locale and have both numbers and values show the correct format
- Users can change currency and have it show in the correct format for that currency
- Users can combine any locale / currency they wish. 
- We don't have custom formats that change based on the size of the number, it can auto-adjust on the fly.
- Because each usage has it's own specification and the same presets aren't used all over the place, it's more easy to change the formatting of one section of Balancer without accidentally breaking other areas. 

In a future PR I'll remove fNum and rename fNum2 to fNum. I didn't want to do it in this PR so that we can ensure all changes are correct first. 

## Type of change


## How should this be tested?

- For each code change, have a look at the relevant unit test and ensure that the new code is the correct equivalent code. 
- Use Balancer and ensure all 

## Checklist:
